code/modules/hallucinations/hallucinations.dm

```diff
@@ -0,0 +1,201 @@
+	HALLUCINATE_MINOR = list(
+		/obj/effect/hallucination/bolts = 10,
+		/obj/effect/hallucination/fake_danger = 10,
+		/obj/effect/hallucination/fake_health = 15,
+		/obj/effect/hallucination/speech = 15,
+		/obj/effect/hallucination/audio = 25,
+		/obj/effect/hallucination/audio/localized = 25,
+	),
+	HALLUCINATE_MODERATE = list(
+		/obj/effect/hallucination/delusion = 5,
+		/obj/effect/hallucination/self_delusion = 5,
+		/obj/effect/hallucination/bolts/moderate = 10,
+		/obj/effect/hallucination/chasms = 10,
+		/obj/effect/hallucination/fake_alert = 10,
+		/obj/effect/hallucination/gunfire = 10,
+		/obj/effect/hallucination/plasma_flood = 10,
+		/obj/effect/hallucination/stunprodding = 10,
+		/obj/effect/hallucination/delamination_alarm = 15,
+		/obj/effect/hallucination/fake_item = 15,
+		/obj/effect/hallucination/fake_weapon = 15,
+		/obj/effect/hallucination/husks = 15,
+	),
+	HALLUCINATE_MAJOR = list(
+		/obj/effect/hallucination/abduction = 10,
+		/obj/effect/hallucination/assault = 10,
+		/obj/effect/hallucination/terror_infestation = 10,
+		/obj/effect/hallucination/loose_energy_ball = 10,
+	)
+))
+
+/mob/living/carbon
+	/// The world.time after which the mob can hallucinate again.
+	var/next_hallucination = 0
+
+/**
+  * Called as part of [/mob/living/proc/handle_status_effects] to handle hallucinations.
+  */
+/mob/living/carbon/proc/handle_hallucinations()
+	if(!hallucination || next_hallucination > world.time)
+		return
+
+	next_hallucination = world.time + rand(HALLUCINATE_COOLDOWN_MIN, HALLUCINATE_COOLDOWN_MAX) / (hallucination * HALLUCINATE_COOLDOWN_FACTOR)
+	if(!prob(HALLUCINATE_CHANCE))
+		return
+
+	// Pick a severity
+	var/severity = HALLUCINATE_MINOR
+	switch(rand(100))
+		if(0 to HALLUCINATE_MINOR_WEIGHT)
+			severity = HALLUCINATE_MINOR
+		if((HALLUCINATE_MINOR_WEIGHT + 1) to HALLUCINATE_MODERATE_WEIGHT)
+			severity = HALLUCINATE_MODERATE
+		if((HALLUCINATE_MINOR_WEIGHT + HALLUCINATE_MODERATE_WEIGHT + 1) to 100)
+			severity = HALLUCINATE_MAJOR
+
+	hallucinate(pickweight(GLOB.hallucinations[severity]))
+
+/**
+  * Spawns an hallucination for the mob.
+  *
+  * Arguments:
+  * * H - The type path of the hallucination to spawn.
+  */
+/mob/living/carbon/proc/hallucinate(obj/effect/hallucination/H)
+	ASSERT(ispath(H))
+	add_attack_logs(null, src, "Received hallucination [H]")
+	return new H(get_turf(src), src)
+
+/**
+  * # Hallucination
+  *
+  * Base object for hallucinations. Contains basic behaviour to display an icon only to the target.
+  */
+/obj/effect/hallucination
+	density = FALSE
+	invisibility = INVISIBILITY_OBSERVER
+	/// Duration in deciseconds. Can also be a list with the form [lower bound, upper bound] for a random duration.
+	var/duration = 15 SECONDS
+	/// Hallucination icon.
+	var/hallucination_icon
+	/// Hallucination icon state.
+	var/hallucination_icon_state
+	/// Hallucination override.
+	var/hallucination_override = FALSE
+	/// Hallucination layer.
+	var/hallucination_layer = MOB_LAYER
```

Comment:
Are these comments really needed? There's no reason *not* to have them of course, but it seems a but superfluous.

Reply:
These could be worded better of course, but they are nonetheless used to generate documentation and it helps to aim for a high coverage to help coders navigate the code more easily.